src/main/java/org/folio/service/routinglists/RoutingListService.java

```diff
@@ -1,51 +1,120 @@
   private static final String ROUTING_LIST_ENDPOINT = resourcesPath(ROUTING_LISTS);
   private static final String ORDER_SETTINGS_ENDPOINT = resourcesPath(ORDER_SETTINGS);
   private static final String ROUTING_USER_ADDRESS_TYPE_ID = "ROUTING_USER_ADDRESS_TYPE_ID";
   private static final String ROUTING_LIST_BY_ID_ENDPOINT = ROUTING_LIST_ENDPOINT + "/{id}";
+  private static final String ROUTING_LIST_BY_POL_ID = "poLineId==%s";
+
   private static final String TEMPLATE_REQUEST_ENDPOINT = resourcesPath(TEMPLATE_REQUEST);
+  private static final UUID TEMPLATE_REQUEST_ID = UUID.fromString("9465105a-e8a1-470c-9817-142d33bc4fcd");
+  private static final String TEMPLATE_REQUEST_LANG = "en";
+  private static final String TEMPLATE_REQUEST_OUTPUT = "text/html";
 
-  private final RestClient restClient;
+  private final PurchaseOrderLineService poLineService;
   private final UserService userService;
+  private final RestClient restClient;
 
-  public RoutingListService(RestClient restClient, UserService userService) {
+  public RoutingListService(RestClient restClient, PurchaseOrderLineService poLineService, UserService userService) {
     this.restClient = restClient;
+    this.poLineService = poLineService;
     this.userService = userService;
   }
 
+  public Future<RoutingList> getRoutingList(String rListId, RequestContext requestContext) {
+    RequestEntry requestEntry = new RequestEntry(ROUTING_LIST_BY_ID_ENDPOINT).withId(rListId);
+    return restClient.get(requestEntry, RoutingList.class, requestContext);
+  }
+
+  public Future<Void> updateRoutingList(RoutingList routingList, RequestContext requestContext) {
+    try {
+      validateRoutingList(routingList, requestContext);
+    }
+    catch (HttpException e) {
+      return Future.failedFuture(e);
+    }
+    RequestEntry requestEntry = new RequestEntry(ROUTING_LIST_BY_ID_ENDPOINT).withId(routingList.getId());
+    return restClient.put(requestEntry, routingList, requestContext);
+  }
+
+  public Future<Void> deleteRoutingList(String rListId, RequestContext requestContext) {
+    RequestEntry requestEntry = new RequestEntry(ROUTING_LIST_BY_ID_ENDPOINT).withId(rListId);
+    return restClient.delete(requestEntry, requestContext);
+  }
+
+  public Future<RoutingList> createRoutingList(RoutingList routingList, RequestContext requestContext) {
+    try {
+      validateRoutingList(routingList, requestContext);
+    } catch (HttpException e) {
+      return Future.failedFuture(e);
+    }
+    RequestEntry requestEntry = new RequestEntry(ROUTING_LIST_ENDPOINT);
+    return restClient.post(requestEntry, routingList, RoutingList.class, requestContext);
+  }
+
+  public Future<RoutingListCollection> getRoutingLists(int limit, int offset, String query, RequestContext requestContext) {
+    RequestEntry requestEntry = new RequestEntry(ROUTING_LIST_ENDPOINT)
+      .withQuery(query)
+      .withOffset(offset)
+      .withLimit(limit);
```

Comment:
would be more readable when preserving the order , if we have parameters limit, offset, then we should declare in the same order in RequestEntry builder or change order in method parameters